### PR TITLE
FM2-12 : Location resource address property definition

### PR DESCRIPTION
**Pull request by jecihjoy**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-12

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**.

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by jecihjoy** [comment]
I'm having some issues using fhir extensions, With my code about when I make a request to the location resource I get this error [https://pastebin.com/2Sb6DRH0](url)

However when I comment out other locations and remain with only one here [https://github.com/jecihjoy/openmrs-module-fhir2/blob/66355528deb0b41c8c53e5fd9fcd693cc47eb057/api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/LocationAddressTranslatorImpl.java#L35](url)

it works fine and this is the response I get [https://pastebin.com/cUrtupED](url)

@ibacher @corneliouzbett  any suggestion on how to fix this?

**Comment by corneliouzbett** [comment]
@jecihjoy Pastebin URLs aren't working for me
I can see neither the error nor response.

**Comment by ibacher** [comment]
@jecihjoy @corneliouzbett I fixed the links in the above comment

**Comment by ibacher** [comment]
@jecihjoy This is my fault for not testing PersonName with more than one field at a time.

In `addAddressExtension()` change the last two lines to:

```java
getOpenmrsAddressExtension(address).orElseGet(() -> address.addExtension().setUrl(FhirConstants.OPENMRS_FHIR_EXT_ADDRESS))
		        .addExtension(FhirConstants.OPENMRS_FHIR_EXT_ADDRESS + "#" + extensionProperty, new StringType(value));
```

What was happening was that a new extension was being created every time `addAddressExtension()` was called, which doesn't matter the first two times (the first one creates the initial extension; the second one creates a new extension; when it gets run a third time, it complains about the two extensions with the same URL).

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/LocationTranslatorImpl.java [hunk 1]
```diff
@@ -19,16 +19,40 @@
...
 public class LocationTranslatorImpl implements LocationTranslator {
 	
+	private LocationAddressTranslatorImpl translator;
+	
 	/**
 	 * @see org.openmrs.module.fhir2.api.translators.LocationTranslator#toFhirResource(org.openmrs.Location)
 	 */
 	@Override
 	public Location toFhirResource(org.openmrs.Location openmrsLocation) {
+		translator = new LocationAddressTranslatorImpl();
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/LocationTranslatorImpl.java]: You can eliminate this line.

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/LocationTranslatorImpl.java [hunk 2]
```diff
@@ -19,16 +19,40 @@
 @Setter(AccessLevel.PACKAGE)
 public class LocationTranslatorImpl implements LocationTranslator {
 	
+	private LocationAddressTranslatorImpl translator;
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/LocationTranslatorImpl.java]: This should be:

```java
@Inject
private LocationAddressTranslator translator;
```

This allows implementations to swap in their own `LocationAddressTranslator` if they need to override the default behaviour.

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/LocationTranslatorImpl.java [hunk 3]
```diff
@@ -19,16 +19,40 @@
...
+	
 	/**
 	 * @see org.openmrs.module.fhir2.api.translators.LocationTranslator#toFhirResource(org.openmrs.Location)
 	 */
 	@Override
 	public Location toFhirResource(org.openmrs.Location openmrsLocation) {
+		translator = new LocationAddressTranslatorImpl();
 		Location fhirLocation = new Location();
+		Location.LocationPositionComponent position = new Location.LocationPositionComponent();
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/LocationTranslatorImpl.java]: Hold off on creating this until we're inside the if block.

**Comment by jecihjoy** [comment]
+1

**Comment by ibacher** [comment]
Nice work!

**Pull request merged**